lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -240,7 +244,54 @@ def convert_string(value: str) -> str:
     return value
 
 
-def convert_access(value: str) -> str:
+def convert_keyword(value: str, parameter: "Parameter") -> str:
+    """
+    Verify that the parameter value is a string and a member of the
+    `valid` list. The match is case-blind and will return the case
+    used in the `valid` list.
+
+    Args:
+        value: parameter value
+        parameter: The Parameter definition (provides valid keywords)
+
+    Raises:
+        ConversionError: input can't be validated or normalized
+
+    Returns:
+        the input value
+    """
+    if type(value) is str:
+        for v in parameter.keywords:
+            if v.casefold() == value.casefold():
+                return v
+    raise ConversionError(value, str(parameter.keywords), str.__name__)
```

Comment:
I think I chose a confusing way to word my comment.  Let me try again....

I think this function should raise `ConversionError` only if the type of `value` is not `str`.  For the unrecognized-keyword case, it should raise a more appropriate/specific exception; if the caller happens to be a List conversion, then it could opt _not_ to catch the exception, which I think would result in the behavior that I would prefer; otherwise, I think it will do the right thing, as well.